finch/processes/utils.py

```diff
@@ -657,6 +657,9 @@ def dataset_to_netcdf(
         for v in ds.data_vars:
             encoding[v] = {"zlib": True, "complevel": compression_level}
 
-    # This is necessary when running with gunicorn
+    # Perform computations
+    ds.load()
+
+    # This is necessary when running with gunicorn to avoid lock-ups
     with dask.config.set(scheduler="single-threaded"):
         ds.to_netcdf(str(output_path), format="NETCDF4", encoding=encoding)
```

Comment:
```suggestion
    ds.to_netcdf(str(output_path), format="NETCDF4", encoding=encoding)
```
If the data is loaded, dask is not involved anymore, and the config set has no effect.